numpy/core/src/multiarray/mapping.c

```diff
@@ -2567,6 +2567,28 @@ PyArray_MapIterCheckIndices(PyArrayMapIterObject *mit)
     NPY_END_THREADS;
     Py_DECREF(intp_type);
     return 0;
+
+indexing_error:
+
+    if (mit->size == 0) {
+        PyObject *err_type = NULL, *err_value = NULL, *err_traceback = NULL;
+        PyErr_Fetch(&err_type, &err_value, &err_traceback);
+        /* 2020-04-02, NumPy 1.19 */
+        if (DEPRECATE(
+                "Invalid index into empty indexing result found. "
```

Comment:
```suggestion
                "Invalid index into empty array found. "
```
The indexing is into the input, not the result, isn't it?